src/main/java/com/epam/esm/utils/QueryConstructor.java

```diff
@@ -10,12 +10,15 @@
     private static final String ORDER_BY_NAME_STATEMENT = " ORDER BY name";
     private static final String ORDER_BY_DATE_STATEMENT = " ORDER BY create_date";
     private static final String DESCENDING_ORDER = " DESC";
+    private static final String COMMA = ", ";
+    private static final String LIMIT = " LIMIT ";
 
-    public String constructQuery(String tagName, String giftCertificateName, String description, String sortByName, String sortByDate) {
-        StringBuilder query = new StringBuilder(SqlQueries.FIND_ALL_CERTIFICATES);
+    public String constructGiftCertificateQuery(String[] tagNames, String giftCertificateName, String description, String sortByName, String sortByDate, Integer page, Integer pageSize) {
```

Comment:
Since we're using JPA & Hibernate now, we should better use Criteria api for building complex requests.

Reply:
Used Criteria API for that purpose